src/routing/kad-dht.md

```diff
@@ -0,0 +1,719 @@
+structured overlay network that allows nodes to locate peers and content in a
+distributed system without relying on centralized servers.
+
+The primary goal of the Kademlia routing algorithm is to progressively discover
+and interact with nodes that are closest to a given key based on the network's
+distance metric. Once a node has identified the closest peers, it can either:
+
+* **Locate a specific peer** in the network
+* **Find content providers** serving content associated with a CID
+* **Store and retrieve values** directly within the DHT, such as IPNS names
+
+### Relation to libp2p kad-dht
+
+The IPFS Kademlia DHT specification is a specialization of the [libp2p Kademlia DHT](https://github.com/libp2p/specs/tree/master/kad-dht).
+
+It is possible to use an alternative DHT specification alongside an IPFS
+implementation, rather than the one detailed here. This document specifically
+outlines all protocol customizations and adaptations required for participation
+in the [Amino DHT](#relation-to-the-amino-dht). If you're designing a new
+Kademlia-based DHT for use with IPFS, some details in this specification may
+appear overly specific or prescriptive.
+
+### Relation to the Amino DHT
+
+Nodes participating in the public [Amino DHT Swarm](#amino-dht) MUST implement the
+IPFS Kademlia DHT specification. The IPFS Kademlia DHT specification MAY be
+used in other DHT swarms as well.
+
+## DHT Swarms
+
+A DHT swarm is a group of interconnected nodes running the IPFS Kademlia DHT
+protocol, collectively identified by a unique libp2p protocol identifier. IPFS nodes
+MAY participate in multiple DHT swarms simultaneously. DHT swarms can be either
+public or private.
+
+### libp2p Protocol Identifier
+
+All nodes participating in the same DHT swarm MUST use the same libp2p protocol
+identifier. The libp2p protocol identifier uniquely identifies a DHT swarm. It
+follows the format `/<swarm-prefix>/kad/<version>`, e.g `/ipfs/kad/1.0.0` for
+the Amino DHT protocol version `1.0.0`, or `/ipfs/lan/kad/1.0.0` for a local
+DHT swarm.
+
+Note that there could be multiple distinct DHT swarms using the same libp2p
+protocol identifier as long as they don't have any common peers. This practice
+is discouraged as networks will immediately merge if they enter in contact.
+
+Each DHT swarm SHOULD have a dedicated protocol identifier.
+
+### Amino DHT
+
+The [Amino DHT](https://blog.ipfs.tech/2023-09-amino-refactoring/#why-amino) is
+the swarm of peers also referred to as the _Public IPFS DHT_. It implements the
+IPFS Kademlia DHT specification and uses the protocol identifier
+`/ipfs/kad/1.0.0`.
+
+:::note
+The Amino DHT is utilized by multiple IPFS implementations, including
+[`kubo`](https://github.com/ipfs/kubo) and
+[`helia`](https://github.com/ipfs/helia)
+and can be joined by using the [public good Amino DHT Bootstrappers](https://docs.ipfs.tech/concepts/public-utilities/#amino-dht-bootstrappers).
+:::
+
+### Client and Server Mode
```

Comment:
Does this make sense in this spec rather than the libp2p one? Is the advice around client / server resource usage not generally applicable?

Reply:
I think this is the similar tradeoff to "RPC messages" – duplicating important content here is intentional, aims to improve readability, creates coherent document, and protects us from feature creep.

The need for jumping from spec HTML into markdown in libp2p/specs repo should be minimized.

Long term, protection from unintended feature creep is important:  someone changes libp2p specs and adds new RPCs, we dont want that to magically apply to Amino DHT.

As prior art example, I copied part of PeerID specs and placed them in https://specs.ipfs.tech/ipns/ipns-record/#ipns-keys to ensure that if libp2p adds new key type, or extends protobuf, that key type does not automatically become part of IPNS specs, but needs an explicit IPIP to add it.